### ISSUE TITLE
Socket is not connected

### ISSUE THREAD

**Issue by vganin** [yandex-cloud/geesefs#63]
Using 0.34.2 on Mac M1 (MacOS Monterey 12.5.1) I'm getting `Socket is not connected` while `fsync`'ing the folder where I copied a bunch of files. Geesefs process is exiting at this point, and I'm loosing the ability to resume uploads after that.

Ran geesefs with options `--debug_s3 --debug_fuse`. Here is the [log](https://github.com/yandex-cloud/geesefs/files/10319456/geesefs-log.txt).

**Comment by vitalif** [comment]
Yeah, the same error is in the log: `writeMessage: socket is not connected [16 0 0 0 0 0 0 0 3 0 0 0 0 0 0 0]`. This means GeeseFS got this error from its fuse (MacFUSE) connection... This probably means MacFUSE dropped the connection. Does macfuse have a log? Can you look in it for errors?

**Comment by vganin** [comment]
Unfortunately I was unable to find any useful logs regarding macfuse anywhere on my Mac. If you have any hint on where to look, I'd really appreciate that.

**Comment by vganin** [comment]
UPDATE: If I'm not using `fsync`, the problem is gone! I tested like 10 times in a row using `fsync` and reproduced problem 100% of time, and then I tried the same test without `fsync` and there was no trouble.

**Comment by vitalif** [comment]
So the error happens on fsync? Interesting, that's at least a direction to look
Does the error happen if you just do 1 fsync?

**Comment by vganin** [comment]
Yes, I'm doing just one single `fsync` on the folder which contains copied files. If that helps, I'm doing `fsync` by means of Rust's `std::fs::File::sync_all` and the debug print of the error is `Os { code: 57, kind: NotConnected, message: "Socket is not connected" }`
